dev/partkey.md

```diff
@@ -0,0 +1,175 @@
+\newcommand \OTSSBatchID {\mathrm{OneTimeSignatureSubkeyBatchID}}
+\newcommand \OneTimeSignature {\mathrm{OneTimeSignature}}
+\newcommand \Sig {\mathrm{Sig}}
+\newcommand \PK {\mathrm{PK}}
+\newcommand \PKSigOld {\mathrm{PKSigOld}}
+\newcommand \PKTwo {\mathrm{PK2}}
+\newcommand \PKOneSig {\mathrm{PK1Sig}}
+\newcommand \PKTwoSig {\mathrm{PK2Sig}}
+
+
+$\OTSSBatchID$ identifies an intermediate level ephemeral sub-key of 
+a batch. $\OTSSBatchID$ is signed by the master key. It has the following
+fields:
+ 
+ - _SubKey Public key_ $\SubKeyPK$, the public key of this sub-key.
+
+ - _Batch_ $\Batch$, batch number of this sub-key.
+
+$\OTSSOffsetID$ identifies an leaf level ephemeral sub-key. $\OTSSOffsetID$
+is signed with a batch sub-key. It has the following fields:
+
+ - _SubKey Public key_ $\SubKeyPK$, the public key of this sub-key.
+
+ - _Batch_ $\Batch$, batch number of this sub-key.
+
+ - _Offset_ $\Offset$, offset of this sub-key in current batch.
+
+Finally, $\OneTimeSignature$ is a cryptographic signature used in voting
+messages between algorand users. It contains the following fields:
+
+ - _Signature_ $\Sig$, a signature of message under $\PK$
+
+ - _Public Key_ $\PK$, the public key of the message signer, 
+   $\PK$ is part of a leaf level ephemeral subkey. 
+
+ - _Old Style Signature_ $\PKSigOld$, **deprecated** field. 
+  It is still in the message only for compability reason.
+
+ - _Public Key 2_  $\PKTwo$, the public key of the current batch. 
+
+ - _Public Key 1 Signature_ $\PKOneSig$, a signature of $\OTSSOffsetID$
+   under $\PKTwo$.
+ 
+ - _Public Key 2 Signature_ $\PKTwoSig$, a signature of $\OTSSBatchID$ under
+   the master key.
+
+VRF Selection Key
+====================
+\newcommand \unauthenticatedVote {\mathrm{unauthenticatedVote}}
+\newcommand \UnauthenticatedCredential {\mathrm{UnauthenticatedCredential}}
+\newcommand \Sender {\mathrm{Sender}}
+\newcommand \Round {\mathrm{Round}}
+\newcommand \Period {\mathrm{Period}}
+\newcommand \Step {\mathrm{Step}}
+\newcommand \Proposal {\mathrm{Proposal}}
+\newcommand \VrfOut {\mathrm{VrfOut}}
+\newcommand \Credential {\mathrm{Credential}}
+\newcommand \Cred {\mathrm{Cred}}
+\newcommand \Weight {\mathrm{Weight}}
+\newcommand \DomainSeparationEnabled {\mathrm{DomainSeparationEnabled}}
+\newcommand \Hashable {\mathrm{Hashable}}
+\newcommand \Vote {\mathrm{Vote}}
+
+To check the validity of a voting message, its VRF selection key
+needs to be verified. Algorand uses Verifiable Random Function (VRF) to 
+generate selection keys (more details in 
+[crypto specification](https://github.com/algorandfoundation/specs/blob/master/dev/crypto.md)).
+
+More specifically, an unverified vote ($\unauthenticatedVote$) has the
+following fields:
+
+ - _Row Vote_ $\mathrm{R}$, an inner struct contains $\Sender$, $\Round$, $\Period$, 
+   $\Step$, and $\Proposal$.
+
+ - _Unverified Credential_ $\Cred$, unverified VRF selection key. $\Cred$ contains 
```

Comment:
Should "unverified VRF selection key" be remove from the sentence?

Reply:
sounds reasonable to me.